core/src/main/java/org/apache/iceberg/FindFiles.java

```diff
@@ -191,7 +191,10 @@ public Builder inPartitions(PartitionSpec spec, List<StructLike> partitions) {
       Snapshot snapshot = snapshotId != null ?
           ops.current().snapshot(snapshotId) : ops.current().currentSnapshot();
 
-      CloseableIterable<ManifestEntry> entries = new ManifestGroup(ops, snapshot.manifests())
+      // snapshot could be null when the table just gets created
+      Iterable<ManifestFile> manifests = (snapshot != null) ? snapshot.manifests() : CloseableIterable.empty();
+
+      CloseableIterable<ManifestEntry> entries = new ManifestGroup(ops, manifests)
```

Comment:
If there are no manifests, then entries should be `CloseableIterable.empty()`, not the manifest iterable. That doesn't need to be closeable.

Reply:
Hi @rdblue , almost got your idea. What about this: when snaphost is null, make collect() return CloseableIterable.empty() directly, as shown in the latest change.

Reply:
Looks good! I'll commit this when tests are passing.